execution/chainspec/genesis.go

```diff
@@ -56,7 +56,7 @@ func ReadPrealloc(fileSys fs.FS, filename string) types.GenesisAlloc {
 // MainnetGenesisBlock returns the Ethereum main net genesis block.
 func MainnetGenesisBlock() *types.Genesis {
 	return &types.Genesis{
-		Config:     MainnetChainConfig,
+		Config:     ReadChainConfig(chainspecs, "chainspecs/mainnet.json"),
```

Comment:
It's probably better to keep `MainnetChainConfig` and refer to it both here and in the `Spec` so that they both share a pointer to the same chain config.